extensions/qcom/GLSL_QCOM_image_processing2.txt

```diff
@@ -0,0 +1,212 @@
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the squares of the values (SSD).     |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchWindowSADQCOM( | Windowed block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSSDQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the squares of the values (SSD).     |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSADQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+
+    * target and reference specify the two input textures. If
+      the block from one texture is going to be used repeatedly across multiple
+      instructions, that texture should be specified as the referenceTexture.
```

Comment:
Is the part about making a texture the referenceTexture a performance note or an actual requirement? I'm not sure what "If the block from one texture ..." really means. Is it when targetCoord/refCoord match? Are the blocks allowed to overlap?

Reply:
yes, it is a performance hint. I updated the text. Let me know if you want it to be further updated.